### ISSUE TITLE
ApiGuard generated different api version for android .so built library

### ISSUE THREAD

**Issue by elpiel** [ralfbiedert/interoptopus#113]
We are working on an app for which we need both .dll and .so to be used in Unity.
For some reason though, the API version differs when built for android and we run the code while the Windows application works totally fine.

Is it possible that for android a different api version is generated?

**Comment by ralfbiedert** [comment]
API guards work like this: 

The signature of all annotated and exported { types, functions, constants } are hashed, including their documentation IIRC.  When writing the `.cs` or interop file, that has is then hard-coded. At the same time, the generated library still has all the information within to recompute its own hash when requested, the same as during API generation.

Normally this should produce the same results between platforms, i.e., "just work". 

Where this might break down is if your exported items or documentation changes between platforms, for example using a platform-dependent type (e.g., `c_int`), or possibly if your Git checks out source code differently between platforms (e.g., if you build the interface on a Windows machine, but the `.so` on a CI; to be honest not sure if this could even produce different results, but worth checking).

**Comment by ralfbiedert** [comment]
Closing for now. Feel free to re-open if this actually turns out to be a bug on our side.

**Comment by elpiel** [comment]
Due to simplicity atm we generate the CS files and use them directly inside the Unity project.

We've tested the shrunk down version and we got different results for the android and Windows version, the ApiVersion differs between Android and Windows platform.

on Windows 650401838260538842
on android 297417645426394928

https://github.com/elpiel/interoptopus-csharp

I've changed the API to try and pinpoint if the issue is one of the callbacks or C# unity side and multi-threading issues so your ApiVersoins might differ.

.dll / .so libraries of the rust core are being build on a Linux machine.
Unity project is being developed on a windows machine.

Dotnet: 6.0
Java: 21
Android (target): aarch64-linux-android
NDK version: 27.0.12077973

`.config.toml`
```
[target.armv7-linux-androideabi]
linker = "armv7a-linux-androideabi21-clang"

[target.aarch64-linux-android]
linker = "aarch64-linux-android21-clang" // needs to point to NDK version + llvm_ar too

[alias]

windows = "build --target x86_64-pc-windows-gnu"
android = "build --target aarch64-linux-android"
```
